crates/consensus/src/transaction/legacy.rs

```diff
@@ -228,6 +228,10 @@ impl Transaction for TxLegacy {
 }
 
 impl SignableTransaction<Signature> for TxLegacy {
+    fn use_eip155(&self) -> bool {
+        self.chain_id.is_some()
```

Comment:
This has the same issue that existed before! Having chain_id does not mean that EIP155 should be used. Having chain_id **in a LEGACY transaction** means EIP155 signature v value should be used.

Reply:
cc @prestwich

Reply:
NVM, Didn't see that this is in legacy file.

Reply:
please read the `impl ... for TxLegacy` immediately above the line you commented on 🙃